src/Server/src/Core/Data/IdentityServer/IdentityServerGroupService.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using IdOps.Model;
+using IdOps.Security;
+using IdOps.Server.Storage;
+
+namespace IdOps
+{
+    public class IdentityServerGroupService : UserTenantService, IIdentityServerGroupService
+    {
+        private readonly IIdentityServerGroupStore _store;
+    
+        public IdentityServerGroupService(
+            IUserContextAccessor userContextAccessor,
+            IIdentityServerGroupStore store)
+            :base(userContextAccessor)
+        {
+            _store = store;
+        }
+    
+        public async Task<IEnumerable<IdentityServerGroup>> GetGroupsByUserTenants(
+            CancellationToken cancellationToken)
+        {
+            IReadOnlyList<string> userTenants =await GetUserTenantsAsync(cancellationToken);
+            IEnumerable<IdentityServerGroup>? allGroups = await GetAllGroupsAsync(cancellationToken);
+    
+            var byTenant = new List<IdentityServerGroup>();
+    
+            foreach (IdentityServerGroup? group in allGroups)
```

Comment:
Why not just `return allGropus.Where(group => group.Tenants.Any(userTenants.Contains).ToList()`

Reply:
i see that this was moved form another file. but we can also clean it up i think

Reply:
Yep, it was only lift and shift

Reply:
Done